packages/auth-construct/API.md

```diff
@@ -47,7 +48,8 @@ export type AuthProps = {
         externalProviders?: ExternalProviderOptions;
     };
     senders?: {
-        email: Pick<UserPoolSESOptions, 'fromEmail' | 'fromName' | 'replyTo'>;
+        email: Pick<UserPoolSESOptions, 'fromEmail' | 'fromName' | 'replyTo'> | IFunction;
+        kmsKeyArn?: string;

```

Comment:
Is `kmsKeyArn` applicable to any of `Pick<UserPoolSESOptions, 'fromEmail' | 'fromName' | 'replyTo'>` ?

If not then we should introduce type for custom sender.
Something like:
`type CustomEmailSender = { handler: IFunction, kmsKeyArn?: string }` and then use that instead of `IFunction` in the union.

Or perhaps have `email: Pick<UserPoolSESOptions, 'fromEmail' | 'fromName' | 'replyTo'> | IFunction | CustomEmailSender` so that passing function without kmskey is still easy.

The bottom line is that we should find typing that doesn't let you specify `kmsKeyArn` if `IFunction` is not used.

@josefaidt wdyt ?

Reply:
this is a good point, i think this would be enough:
email: Pick<UserPoolSESOptions, 'fromEmail' | 'fromName' | 'replyTo'> | CustomEmailSender

Reply:
feedback makes sense. I like this take
```
email: Pick<UserPoolSESOptions, 'fromEmail' | 'fromName' | 'replyTo'> | IFunction | CustomEmailSender
```
this is also simple, and easier to understand that you only have two valid configuration options (ses or a sender function with/without a custom kms key)
```
email: Pick<UserPoolSESOptions, 'fromEmail' | 'fromName' | 'replyTo'> | CustomEmailSender
```